### PR TITLE
noise: use locking instead of atomics

### PR DESCRIPTION
**Pull request by Noah-Kennedy**

### CODE REVIEW

File: boringtun/src/noise/mod.rs [hunk 1]
```diff
@@ -172,7 +172,9 @@ impl Tunn {
         static_public: x25519_dalek::PublicKey,
         rate_limiter: Option<Arc<RateLimiter>>,
     ) -> Result<(), WireGuardError> {
-        self.timers.should_reset_rr = rate_limiter.is_none();
+        let mut timers = self.timers.lock();
```
Comment (jeff-hiner) [inline boringtun/src/noise/mod.rs:175]: Tightly scope this in a block to rule out potential deadlocks.

File: boringtun/src/noise/mod.rs [hunk 2]
```diff
@@ -378,14 +380,17 @@ impl Tunn {
 
     /// Update the index of the currently used session, if needed
     fn set_current_session(&self, new_idx: usize) {
+        let timers = self.timers.lock();
```
Comment (jeff-hiner) [inline boringtun/src/noise/mod.rs:383]: Move this below the return to reduce lock scope/duration.

**Comment by zaid3tech** [comment]
@Noah-Kennedy will this solve the `session being dropped issue`, discussed here #258

**Comment by Noah-Kennedy** [comment]
@zaid3tech It should fix that, yes. As is this is WIP because I need to restructure a bunch of stuff to get things to not deadlock

**Pull request closed without merging**